pulsar-common/src/main/java/org/apache/pulsar/common/topics/TopicList.java

```diff
@@ -56,6 +57,12 @@ public static List<String> filterTopics(List<String> original, Pattern topicsPat
                 .collect(Collectors.toList());
     }
 
+    public static List<String> filterTransactionInternalName(List<String> original) {
+        return original.stream()
+                .filter(topic -> !SystemTopicNames.isTransactionInternalName(TopicName.get(topic)))
```

Comment:
Why just filter transaction internal name? Something like(`__transaction_buffer_snapshot`, `__transaction_buffer_snapshot_segments`, `__transaction_buffer_snapshot_indexes`) these doesn't seem to be filtered out